pkg/image/graph/edges.go

```diff
@@ -24,11 +24,25 @@ func AddImageStreamRefEdge(g osgraph.MutableUniqueGraph, node *imagegraph.ImageS
 	g.AddEdge(node, imageStreamNode, ReferencedImageStreamGraphEdgeKind)
 }
 
-// AddAllImageStreamRefEdges calls AddImageStreamRefEdge for every ImageStreamTagNode in the graph
+// AddImageStreamImageRefEdge ensures that a directed edge exists between an ImageStreamImage Node and the IS it references
+func AddImageStreamImageRefEdge(g osgraph.MutableUniqueGraph, node *imagegraph.ImageStreamImageNode) {
+	dockImgRef, _ := imageapi.ParseDockerImageReference(node.Name)
+	imageStream := &imageapi.ImageStream{}
+	imageStream.Namespace = node.Namespace
+	imageStream.Name = dockImgRef.Name
+
+	imageStreamNode := imagegraph.FindOrCreateSyntheticImageStreamNode(g, imageStream)
+	g.AddEdge(node, imageStreamNode, ReferencedImageStreamGraphEdgeKind)
```

Comment:
It seemed like a less invasive change.  And other than what the PR added in bc.go, the other examiners of that edge kind (image_pipeline.go, dc.go) did not need to distinguish between ImageStreamTag's and ImageStreamImage's.

Your point is valid.  I'll add a new edge kind (and rename the existing one to better distinguish between ImageStreamImage and ImageStreamTag, unless you disagree) and swizzle image_pipeline.go, dc.go, and bc.go to use them.  Among other things, it probably will no-op the break / continue look stuff we've discussed here, simplifying things.